### DISCUSSION TITLE
RedisException

### DISCUSSION THREAD

**Discussion by punkyard** [nextcloud/all-in-one#5225] [Questions] [answered]
hi
I'm getting a recurrent error in the nextcloud.log

{"Exception":"RedisException","Message":"Redis server nextcloud-aio-redis:6379 went away","Code":0,

but all my containers are green in AiO interface ..

#### Output of `sudo docker info`
```
Client: Docker Engine - Community
 Version:    27.2.0
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.2
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
    ..
    Server Version: 27.2.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 472731909fa34bd7bc9c087e4c27943f9835f111
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.10.0-30-amd64
 Operating System: Debian GNU/Linux 11 (bullseye)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.759GiB
 Name: rez0.net
 ID: c7e9ce77-e3f3-4bee-87bc-5718291412ca
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

#### Docker run command or docker-compose file that you used
`tail -n 100 /var/www/html/data/nextcloud.log`

#### Other valuable info 
`docker restart nextcloud-aio-nextcloud` renders:
```
Memory overcommit is disabled but necessary for safe operation
See https://github.com/nextcloud/all-in-one/discussions/1731 how to enable overcommit
Redis has started
6:C 01 Sep 2024 06:01:31.292 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
6:signal-handler (1725213747) Received SIGTERM scheduling shutdown...
6:M 01 Sep 2024 20:02:27.862 # Redis is now ready to exit, bye bye...
Memory overcommit is disabled but necessary for safe operation
See https://github.com/nextcloud/all-in-one/discussions/1731 how to enable overcommit
Redis has started
8:C 01 Sep 2024 20:02:42.814 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
```

I will try to add two lines to my yaml:
```
     - REDIS_SYSCTLS=vm.overcommit_memory=1
     - OVERWRITEPROTOCOL=https
```

is this an issue?

thank you very much for your attention

**Comment by szaimen** [comment]
> Memory overcommit is disabled but necessary for safe operation
> See https://github.com/nextcloud/all-in-one/discussions/1731 how to enable overcommit

Hi, can you follow this documentation and check if it helps?

**Comment by szaimen** [comment]
Hi @punkyard any update here?

**Comment by punkyard** [comment]
Thanks Simon for coming back to me as it seems a bit confusing to me

it seems i had it wrong from the start.. as I couldn't find '/etc/sysctl.d/nextcloud-aio-memory-overcommit.conf' on the host, I ran `sudo sysctl vm.overcommit_memory=1` as it was written in the error message.

Now, i don't understant how i should applied the command advised in [the](https://github.com/nextcloud/all-in-one/discussions/1731):  `echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf'?

1. straight, with the sudo user of the host?
    $ echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf
    or
    $ sudo echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf
2. inside the container with exec? but in which container? nextcloud-aio-nextcloud? like:
    `docker exec -u www-data -it nextcloud-aio-nextcloud /bin/bash'
    `echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf`


I'm asking cause i'm afraid to mess up - this is still beyond my fields
thanks a lot for your precious insight

**Comment by szaimen** [comment]
I see 👍

> 1. straight, with the sudo user of the host?
> echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf

This should work.

**Comment by punkyard** [comment]
ok, so it's just this one line to be added to the freshly created nextcloud-aio-memory-overcommit.conf in /etc/sysctl.d

i guess then I will have to wait a bit to see if the cascade of errors are related or not:

`DbalException An exception occurred while executing a query: SQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint "files_versions_uniq_index" DETAIL: Key (file_id, "timestamp")=(301672, 1725872875) already exists. Exception thrown: OC\DB\Exceptions\DbalException`

`RuntimeExceptionOCA\Files_Antivirus\BackgroundJob\BackgroundScanner::processFiles, exception:OCA\Files_Antivirus\BackgroundJob\BackgroundScanner::processFiles, exception:`

`[PHP] Error: unserialize(): Error at offset 40 of 43 bytes at /var/www/html/apps/dav/lib/DAV/CustomPropertiesBackend.php#574 PROPFIND /remote.php/dav/addressbooks/users/username/`

I will come back with an update - thanks a lot Simon

**Comment by szaimen** [comment, marked as answer]
The issues that you are now reporting are unrelated to the redis issue.

> `DbalException An exception occurred while executing a query: SQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint "files_versions_uniq_index" DETAIL: Key (file_id, "timestamp")=(301672, 1725872875) already exists. Exception thrown: OC\DB\Exceptions\DbalException`

This should be reported to https://github.com/nextcloud/server/issues if not already.

> `RuntimeExceptionOCA\Files_Antivirus\BackgroundJob\BackgroundScanner::processFiles, exception:OCA\Files_Antivirus\BackgroundJob\BackgroundScanner::processFiles, exception:`

This should be reported to https://github.com/nextcloud/files_antivirus/issues if not already.

> `[PHP] Error: unserialize(): Error at offset 40 of 43 bytes at /var/www/html/apps/dav/lib/DAV/CustomPropertiesBackend.php#574 PROPFIND /remote.php/dav/addressbooks/users/username/`

This should also be reported to https://github.com/nextcloud/server/issues if not already.

**Comment by punkyard** [comment]
ok i will
all these errors appeared at the same time with the redis error.
have a good day Simon 🙏

**Reply by szaimen** [reply]
> all these errors appeared at the same time with the redis error.

I see. 

> have a good day Simon 🙏

Thanks, you too! :)
